org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -1321,7 +1349,14 @@ class CodeActionContext {
  * These commands are typically code fixes to either fix problems or to beautify/refactor code.
  */
 @JsonRpcData
-class CodeActionParams {
+class CodeActionParams implements WorkDoneProgressParams, PartialResultParams {
+	
+	@JsonAdapter(EitherTypeAdapter.Factory)
```

Comment:
EitherTypeAdapter is registered globally in Gson. You should be able to remove these annotations and it should work the same.